### PR TITLE
Add 'DB Instance Publicly Accessible' query for AWS Ansible

### LINKED ISSUE #1263: Add 'DB Instance Publicly Accessible' query for AWS Ansible
### Platform
*Ansible*

### Provider
*AWS*

### Description
*RDS Instance publicly_accessible flag should not be true*

### PR DESCRIPTION
**Pull request by rogeriopeixotocx**
### Platform
*Ansible*

### Provider
*AWS*

### Description
*RDS Instance publicly_accessible flag should not be true*

Closes #1263

### CODE REVIEW

File: assets/queries/ansible/aws/db_instance_publicly_accessible/query.rego [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    document := input.document[i]
+    tasks := getTasks(document)
+    task := tasks[t]
+    isAnsibleTrue(task["community.aws.rds_instance"].publicly_accessible)
+    result := {
+        "documentId": document.id,
+        "searchKey": sprintf("name=%s.{{community.aws.rds_instance}}.publicly_accessible", [task.name]),
+        "issueType": "IncorrectValue",
+        "keyExpectedValue": "community.aws.rds_instance.publicly_accessible should be false",
+        "keyActualValue": "community.aws.rds_instance.publicly_accessible is true"
```
Comment (joaomartinscx) [inline assets/queries/ansible/aws/db_instance_publicly_accessible/query.rego:13]: Should specify the resource name to help pinpoint the failing resource

File: assets/queries/ansible/aws/db_instance_publicly_accessible/query.rego [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    document := input.document[i]
+    tasks := getTasks(document)
+    task := tasks[t]
+    isAnsibleTrue(task.rds.publicly_accessible)
+    result := {
+        "documentId": document.id,
+        "searchKey": sprintf("name=%s.rds.publicly_accessible", [task.name]),
+        "issueType": "IncorrectValue",
+        "keyExpectedValue": "aws.rds.publicly_accessible should be false",
+        "keyActualValue": "aws.rds.publicly_accessible is true"
```
Comment (joaomartinscx) [inline assets/queries/ansible/aws/db_instance_publicly_accessible/query.rego:27]: The collections Amazon.aws does not seem to have this module https://docs.ansible.com/ansible/latest/collections/amazon/aws/index.html#plugins-in-amazon-aws

**Review by joaomartinscx — CHANGES_REQUESTED** [review]

File: assets/queries/ansible/aws/db_instance_publicly_accessible/query.rego (hunk 2 above)
Comment (rogeriopeixotocx) [reply assets/queries/ansible/aws/db_instance_publicly_accessible/query.rego:27]: Removed `aws`
example: https://github.com/AutomationWithAnsible/ansible-aws-rds/blob/master/tasks/rds.yml#L4-L5

File: assets/queries/ansible/aws/db_instance_publicly_accessible/query.rego (hunk 1 above)
Comment (rogeriopeixotocx) [reply assets/queries/ansible/aws/db_instance_publicly_accessible/query.rego:13]: we can specify only the task name, not the resource name, but ok. I'll put in the task name

File: assets/queries/ansible/aws/db_instance_publicly_accessible/query.rego (hunk 2 above)
Comment (joaomartinscx) [reply assets/queries/ansible/aws/db_instance_publicly_accessible/query.rego]: the module doesn't exist in the docs, the link mentioned in that example is not found (404) in the docs

File: assets/queries/ansible/aws/db_instance_publicly_accessible/query.rego [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    task := tasks[t]
+    isAnsibleTrue(task.rds.publicly_accessible)
+    result := {
+        "documentId": document.id,
+        "searchKey": sprintf("name=%s.rds.publicly_accessible", [task.name]),
+        "issueType": "IncorrectValue",
+        "keyExpectedValue": sprintf("module's parameter rds.publicly_accessible should be false in task: '%s'", [task.name]),
+        "keyActualValue": sprintf("module's parameter rds.publicly_accessible is true in task: '%s'", [task.name])
+    }
+}
```
Comment (joaomartinscx) [inline assets/queries/ansible/aws/db_instance_publicly_accessible/query.rego:29]: This resource doesn't exist, the doc for this module is no longer in the docs: https://docs.ansible.com/ansible/latest/rds_module.html

**Review by joaomartinscx — CHANGES_REQUESTED** [review]

**Comment by markmishaevcx** [comment]
Scan submitted to Checkmarx

File: assets/queries/ansible/aws/db_instance_publicly_accessible/query.rego (hunk 3 above)
Comment (rogeriopeixotocx) [reply assets/queries/ansible/aws/db_instance_publicly_accessible/query.rego]: You're right. Good catch. Removing this section and sample.

**Review by joaomartinscx — APPROVED** [review]
Everything looks okay, good job!

**Comment by markmishaevcx** [comment]
Logo<br />**Checkmarx SAST - Scan Summary & Details**
### Cx-SAST Summary
Total of 4 vulnerabilities<br />High&nbsp;**0 High**<br />Medium&nbsp;**3 Medium**<br />Low&nbsp;**1 Low**<br />Info&nbsp;**0 Info**<br />
### Violation Summary
Medium&nbsp;**3** **Medium**<br />
[View more details on Checkmarx UI](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049039&projectid=719)
### Cx-SAST Details
|Lines|Severity|Category|File|Link|
---|---|---|---|---
[30](https://github.com/Checkmarx/kics/blob/feature/db_instance_publicly_accessible/cmd/builder/main.go#L30) |Medium|Path_Traversal|cmd/builder/main.go|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049039&projectid=719&pathid=3)
[52](https://github.com/Checkmarx/kics/blob/feature/db_instance_publicly_accessible/pkg/kics/service.go#L52) |Medium|Denial_Of_Service_Resource_Exhaustion|pkg/kics/service.go|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049039&projectid=719&pathid=1)
[82](https://github.com/Checkmarx/kics/blob/feature/db_instance_publicly_accessible/test/main_test.go#L82) |Medium|Denial_Of_Service_Resource_Exhaustion|test/main_test.go|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049039&projectid=719&pathid=2)

**Pull request merged**